### DISCUSSION TITLE
Can we get separate AspNetCore metrics for each Controller-Action?

### DISCUSSION THREAD

**Discussion by musaul** [open-telemetry/opentelemetry-dotnet#3154] [Q&A]
I'm using `OpenTelemetry.Instrumentation.AspNetCore` version `1.0.0-rc9.1`, and using the default AspNetCore metrics (i.e. `AddOpenTelemetryMetrics` ... `AddAspNetCoreInstrumentation`) I get separate histograms for each http _flavour_, _scheme_, and _status-code_ combination. This seems to be based on the tags in [HttpInMetricsListener.cs](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInMetricsListener.cs).

Is there currently (or in the plan) any way of adding more tags to add more dimensions to the metrics, or just get completely seperate sets of metrics.  Ideally, as the title suggests, I'd like separate histograms for each Controller Action. If there is a way to add the http.uri tag/dimension, then that could also work as a stop-gap.

**Comment by alanwest** [comment]
There is an open issue for adding additional dimensions which may satisfy your use case https://github.com/open-telemetry/opentelemetry-dotnet/issues/3083.
